packages/realm-server/server.ts

```diff
@@ -601,6 +603,51 @@ export class RealmServer {
 
     throw new Error(`Can not determine the matrix registration secret`);
   }
+
+  private fetchCatalogRealms(): (
+    ctxt: Koa.Context,
+    _next: Koa.Next,
+  ) => Promise<any> {
+    return async (ctxt: Koa.Context, _next: Koa.Next) => {
+      let results = await fetchPublicRealms(this.dbAdapter);
+      let promises = results.map(async ({ realm_url: realmURL }) => {
+        let response = await this.virtualNetwork.handle(
+          new Request(`${realmURL}_info`, {
+            headers: {
+              Accept: SupportedMimeType.RealmInfo,
+            },
+          }),
+        );
+        if (response.status != 200) {
+          console.warn(
+            `Failed to fetch realm info for public realm ${realmURL}: ${response.status}`,
+          );
+          return;
+        }
+        let json = await response.json();
+        let attributes = json.data.attributes;
+        if (
+          attributes.showAsCatalog != null &&
+          attributes.showAsCatalog == false
+        ) {
+          return;
+        }
+
+        return {
+          type: 'catalog-realm',
+          id: realmURL,
+          attributes: json.data.attributes,
+        };
+      });
+      let data = (await Promise.all(promises)).filter(Boolean);
```

Comment:
it would be nice to cache this answer. right now its a big huge ordeal to make a new public catalog. it's incredibly unlikely that one will be made on the fly. So let's just calculate this answer once and cache it. In the future when it's possible to make public catalogs on the fly we can revisit this.